### PR TITLE
Fix getter/setter deprecation warnings

### PR DESCRIPTION
**Pull request by Gaurav0**
Fixes a lot of annoying getter/setter deprecation warnings.

This uses the ember-new-computed addon to fix the getter/setter deprecation warnings in a manner that does not break Ember versions less than 1.11

### CODE REVIEW

File: addon/components/ember-table.js [hunk 1]
```diff
@@ -137,29 +138,33 @@ StyleBindingsMixin, ResizeHandlerMixin, {
...
+        case 'single':
+          return selection[0] || null;
+        case 'multiple':
+          return selection;
+      }
+    },
+    set: function(key, value) {
+      var selectionMode = this.get('selectionMode');
       this.get('persistedSelection').clear();
       this.get('rangeSelection').clear();
```
Comment (cyjia) [inline addon/components/ember-table.js]: Do we need to test `value` null or undefined?

**Comment by cyjia** [comment]
:+1:

File: addon/components/ember-table.js (hunk 1 above)
Comment (Gaurav0) [inline addon/components/ember-table.js]: I'm not sure but code was not doing so previously.
